crucible-llvm/src/Lang/Crucible/LLVM/Translation.hs

```diff
@@ -389,18 +361,18 @@ translateModule :: (?laxArith :: Bool)
                 -> L.Module        -- ^ Module to translate
                 -> IO (Some ModuleTranslation)
 translateModule halloc m = do
-  Some ctx0 <- mkLLVMContext halloc m
+  Some ctx <- mkLLVMContext halloc m
   let nonceGen = haCounter halloc
-  llvmPtrWidth ctx0 $ \wptr -> withPtrWidth wptr $
-    do -- Add handles for all functions declared in module.
-       ctx <- foldM (insDeclareHandle halloc) ctx0 (allModuleDeclares m)
-       -- Translate definitions
-       pairs <- mapM (transDefine ctx) (L.modDefines m)
-       -- Return result.
+  llvmPtrWidth ctx $ \wptr -> withPtrWidth wptr $
+    do pairs <- mapM (transDefine halloc ctx) (L.modDefines m)
+
        let ?lc  = ctx^.llvmTypeCtx -- implicitly passed to makeGlobalMap
+       let ctx' = ctx{ llvmGlobalAliases = globalAliases m
+                     , llvmFunctionAliases = functionAliases m
+                     }
        nonce <- freshNonce nonceGen
        return (Some (ModuleTranslation { cfgMap = Map.fromList pairs
                                        , globalInitMap = makeGlobalMap ctx m
-                                       , _transContext = ctx
+                                       , _transContext = ctx'
```

Comment:
Previously `_transContext` was the same context passed to `globalInitMap`. Why are they different now? Can we not pass `ctx'` to `makeGlobalMap`?

Reply:
I don't think there's a reason, I just missed changing the one passed to `makeGlobalMap`.  I don't think it makes  use of the alias maps, so it shouldn't make a difference, but it's probably better to make them the same.

I could also be convinced that these alias maps should be cached elsewhere instead (say in the `ModuleTranslation`), I'm not really sure.